.github/workflows/ci.yml

```diff
@@ -10,6 +10,17 @@ on:
   workflow_call:
 
 jobs:
+  test_go:
+    name: Go tests
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/prometheus/golang-builder:1.23-base
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: prometheus/promci@c3c93a50d581b928af720f0134b2b2dad32a6c41 # v0.4.6
+      - uses: ./.github/promci/actions/setup_environment
+      - run: go test ./pkg/...
```

Comment:
Maybe just `make test` could do?

Reply:
We should also extend the `needs` in the `publish_{master,release}` steps later on to depend on this job.

Reply:
> We should also extend the `needs` in the `publish_{master,release}` steps later on to depend on this job.

@jeschkies can you please address this other comment?